File: content/atlas/source/includes/fts/field-types/embeddedDocuments/create-index-basic.js
const { MongoClient } = require("mongodb");

// Replace the connection string with your MongoDB deployment's connection string
const uri =  "<connection-string>";

const client = new MongoClient(uri);

async function run() {
  try {
    const database = client.db("sample_supplies");
    const collection = database.collection("sales");

    // Create the Atlas Search index definition for the embeddedDocuments field
    const index = {
        name: "default",
        definition: {
            "mappings": {
                "dynamic": false,
                "fields": {
                    "items": {
                        "type": "embeddedDocuments",
                        "dynamic": true
                    }
                }
            }
        }
    }

    // Create the index
    const result = await collection.createSearchIndex(index);
    console.log("New index name: " + result);
  } finally {
    await client.close();
  }
}

run().catch(console.dir);

File: content/atlas/source/includes/fts/field-types/embeddedDocuments/create-index-basic.java
import com.mongodb.client.*;
import org.bson.Document;

public class CreateIndex {
    public static void main(String[] args) {
        // Replace the connection string with your MongoDB deployment's connection string
        String connectionString = "<connection-string>";

        try (MongoClient mongoClient = MongoClients.create(connectionString)) {
            MongoDatabase database = mongoClient.getDatabase("sample_supplies");
            MongoCollection<Document> collection = database.getCollection("sales");
            String indexName = "default";

            // Create the MongoDB Search index definition for the embeddedDocuments field
            Document searchIdx = new Document(
                    "mappings",
                    new Document("dynamic", false)
                            .append("fields",
                                    new Document("items",
                                            new Document("type", "embeddedDocuments")
                                                .append("dynamic", true)))
            );
            collection.createSearchIndex(indexName, searchIdx);
            System.out.println("New index name: " + indexName);
        } catch (Exception e) {
            System.err.println("Error creating Atlas Search index: " + e.getMessage());
        }
    }
}